CONTRIBUTING.md

```diff
@@ -1,7 +1,21 @@
-# Contributing to thorn
+# Contributing to Thorn
 
 Think twice, code once.
 
+Pull requests are accepted under the discretion of Thorn maintainer(s). Improve the chances of your pull requests getting merged by following the requirements and guidelines below. For any questions, please e-mail developers@sugarcrm.com.
+
+## Requirements
+- By creating a pull request, you are agreeing to [SugarCRM Open Source Contributor terms](CONTRIBUTOR_TERMS.pdf).
+
+## Guidelines
+- Make sure your pull request contains quality code. We will certainly provide constructive feedback on works in progress but we will not merge incomplete pull requests.
+- Make sure your pull requests are properly tested and maintain (or improve) the current code coverage.
+- Make sure your pull request is fully documented.
+- Reference related Github issues within commit messages and pull request comments where appropriate.
```

Comment:
@yvan33 and @rhoggSugarcrm totally agree, my point is:
when we do `yarn install` using node 7 we get stuff like
```
warning gulp > vinyl-fs > glob-stream > minimatch@2.0.10: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
warning gulp > vinyl-fs > glob-watcher > gaze > globule > minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
warning gulp > vinyl-fs > glob-watcher > gaze > globule > glob > graceful-fs@1.2.3: graceful-fs v3.0.0 and before will fail on node releases >= v7.0. Please update to graceful-fs@^4.0.0 as soon as possible. Use 'npm ls graceful-fs' to find it in the tree.
```
this means that we will need to keep upgrading these libs in order to avoid these warnings or even errors... the less dependencies we have, the better (like you both said) 😉